[OSD-7582] Refactor makefile  and dev improvement (#64)
openshift/aws-account-shredder: README.md

Assert that you have no failed accounts in the `aws-account-operator` namespace, otherwise these will be shredded once you run the next step:
```
oc -n aws-account-operator get accounts
```

Then, deploy the account shredder by running:
```
make deploy
```

## Running an ad-hoc shred

The following steps will perform the shredding on the account-id specified earlier. ( the `ACCOUNT_ID_TO_SHRED` environment variable ).

> DOUBLE CHECK THAT YOU'VE CONFIGURED THE CORRECT ACCOUNT ID, AS THIS IS A DESTRUCTIVE OPERATION THAT CAN NOT BE UNDONE!

Run the following to create the account:
```
make create-account
```

This will use the template [aws-shredder-account-delete-template.yaml](./hack/templates/aws-shredder-account-delete-template.yaml) to create a new account cr that represents your aws account. Then, use the next command to mark the account as `failed`, which will cause the account shredder to shred that account:

```
make shred-account
```

Once you set the status of the account to failed, the Shredder should pick it up and start shredding through the accounts.

> You should be able to follow the logs and watch the shred happen using `make get-logs`.  Certain objects may not delete on the first attempt through the shredder, but the shredder will continue to run on the account until it is created.

Reviewer: ```suggestion
You should be able to follow the logs and watch the shred happen using `make get-logs`.  Certain aws resources may not completely delete on the first attempt through the shredder, but the shredder will continue to run on the account until it is cleaned.
```

```
docker push quay.io/<username>/aws-account-shredder:<tag>
```

> Now that you have your image pushed up, you can simply apply the deployment again using `make deploy`

Reviewer 2: does the user need to set a IMG env variable or something like that so the make will use the correct image?

Reviewer 2: ```suggestion
Now that you have your image pushed up, you can apply the deployment again using `make deploy`
```
words like simply or easy should be banned from all documentation. What's simple for you, can be complicated for someone else ;)

Author: > does the user need to set a IMG env variable or something like that

Setting the image repository is documented a couple of lines above the one i've changed, so it did not appear in the diff:
```
You'll need to edit the `IMAGE_REPOSITORY` field in `project.mk` to point at your `aws-account-shredder` repo 
```

Author: Alternatively, ( and probably better ), one can override the variable on the commandline or with an env var:
```
make docker-build IMAGE_REPOSITORY=maschulz
```
and
```
export IMAGE_REPOSITORY=maschulz
make docker-build
```
achieve exactly the same. Do you have a preference for what should be documented as the "default" way?

Author: I added a line to the readme to clarify this a bit.

> Now that you have the your image pushed up, you can simply apply the deployment again using `make deploy`

Reviewer: ```suggestion
Now that you have your image pushed up, you can simply apply the deployment again using `make deploy`
```
